Detectors/TRD/simulation/src/Digitizer.cxx

```diff
@@ -459,9 +459,8 @@ bool Digitizer::convertSignalsToADC(const int det, int& signals)
   float calGainFactorDetValue = 0.47; // +/- 0.06 // Defaults value  from OCDB (AliRoot DrawTrending macro) for 5 TeV pp - 27 runs from LHC15n
 
   // Create the digits for this chamber
-  for (row = 0; row < nRowMax; row++) {
-    for (col = 0; col < nColMax; col++) {
-
+  for (int row = 0; row < nRowMax; row++) {
+    for (int col = 0; col < nColMax; col++) {
```

Comment:
Here it should be more efficient to loop over the keys in the adcMapCont. 

This loop will touch many pads that do not have data, and you lookup gain factors etc...

Reply:
You are right. I will change this double nested loop by a single loop over the existing elements in the map only.